### ISSUE TITLE
Function/Event to get a download status for the PDF file

### ISSUE THREAD

**Issue by vic1707** [stephanrauh/ngx-extended-pdf-viewer#588]
**Is your feature request related to a problem? Please describe.**
Being able to get at any time (or at least frequently) the current download state of the PDF file, in order to be able to make a download bar.
The Function/Event would from an '$event' argument give the number of currently downloaded bytes and the total bytes of the file. So the user can make a download bar besed on the percentage obtained from those number or anything he wants.

It would be the same feature as the '(on-progress)' present on 'ng2-pdf-viewer' (which unfortunately doesn't provide the total size of the PDF file).

**Describe the solution you'd like**
A simple function binded from the angular component : 

`(Progress)="Progress($event)"`

**Describe alternatives you've considered**
There is a hack described on an issue but not as clean as being able to have it directly from 'ngx-extended-pdf-viewer'

**Comment by stephanrauh** [comment]
I assume you need a progress bar when the PDF file is loaded? I've checked the "download" function, but it don't think a progress bar makes sense when downloading files. The file is already in memory, so the download doesn't take long.

However, I can implement progress events for printing and for loading.

**Comment by vic1707** [comment]
When using the `ng2-pdf-viewer` I used `(Progress)="Progress($event)"` in order to display by myself (via the progress tag) the current state of the download, in the same way as the chrome console in the network tab.
```  
<progress
    max="{{ sizePDF }}"
    value="{{ sizeLoadedPDF }}"
  ></progress>
```
It was made to know that the file is still downloading, estimate (by eye not by code) the remaining time and be able to customize it.

**Comment by stephanrauh** [comment]
The new event `(progress)` has landed with version 7.2.0.

Enjoy!
Stephan
